apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -527,3 +529,40 @@ export async function findByStatus({
     throw error;
   }
 }
+
+/**
+ * Find the best placement of an agent across all competitions
+ * @param agentId The agent ID
+ * @returns The agent best placement
+ */
+export async function findBestPlacementForAgent(agentId: string) {
```

Comment:
this is where we use the new `competitions_leaderboard` table. this new table doesnt have any logic that writes to it, yet, so the tests in the PR operate with this assumption.

the `findBestPlacementForAgent` will query the `competitions_leaderboard` table for the top "rank" by agent ID. then, it gets the total # of agents that were in that competitions so that we can show the agent placed _N_ of _M_.